packages/@aws-cdk/aws-lambda-python/lib/function.ts

```diff
@@ -113,7 +113,7 @@ export class PythonFunction extends lambda.Function {
         assetHashType: props.assetHashType,
         assetHash: props.assetHash,
       }),
-      handler: `${index.slice(0, -3)}.${handler}`,
+      handler: `${index.replace(/\.py$/i, '').replace('/', '.')}.${handler}`,
```

Comment:
Can you add a little comment with a small example explaining this string replace?